### PR TITLE
Feature/mat 6902 hr hqmf performance

### PR DESCRIPTION
**Pull request by nmorasb**
## MADiE PR

Jira Ticket: [MAT-6902](https://jira.cms.gov/browse/MAT-6902)
(Optional) Related Tickets:

### Summary
Move QDM Human Readable generation into madie-qdm-service.
This allows us to fetch the needed data (CqlLookups) only one time to generate both HQMF and HR for the measure package.

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by jkotanchik-SB — APPROVED** [review]

File: src/main/java/gov/cms/madie/services/HumanReadableService.java [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+@Slf4j
+public class HumanReadableService {
+
+  private Template baseHumanReadableTemplate;
+  private final Collator collator = Collator.getInstance(Locale.US);
+
+  /**
+   * Generates the QDM Human Readable HTML from a MADiE Measure.
+   *
+   * @param measure MADiE Measure
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/services/HumanReadableService.java:56]: New param could use a javadoc entry.

File: src/main/java/gov/cms/madie/dto/SourceDataCriteria.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package gov.cms.madie.dto;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class SourceDataCriteria implements Comparable<SourceDataCriteria> {
```
Comment (adongare) [inline src/main/java/gov/cms/madie/dto/SourceDataCriteria.java:8]: Still required?

File: src/main/java/gov/cms/madie/services/HumanReadableService.java (hunk 1 above)
Comment (nmorasb) [reply src/main/java/gov/cms/madie/services/HumanReadableService.java:56]: will update javadoc in a subsequent PR

File: src/main/java/gov/cms/madie/dto/SourceDataCriteria.java (hunk 2 above)
Comment (nmorasb) [reply src/main/java/gov/cms/madie/dto/SourceDataCriteria.java:8]: will remove in a subsequent PR
